client/components/chart/legend.js

```diff
@@ -36,65 +37,108 @@ function getFormatedTotal( total, valueType ) {
  * A legend specifically designed for the WooCommerce admin charts.
  */
 class Legend extends Component {
+	constructor() {
+		super();
+
+		this.listRef = createRef();
+
+		this.state = {
+			isScrollable: false,
+		};
+	}
+
+	componentDidMount() {
+		this.updateListScroll();
+		window.addEventListener( 'resize', this.updateListScroll );
+	}
+
+	componentWillUnmount() {
+		window.removeEventListener( 'resize', this.updateListScroll );
+	}
+
+	updateListScroll = () => {
+		const list = this.listRef.current;
```

Comment:
does react allow for request animation frame?

Reply:
I found something in the docs that might be useful in this particular case:
https://reactjs.org/docs/faq-functions.html#requestanimationframe-throttling

I spent some time playing with it but I don't have anything ready yet. I created an issue so we can take a look at it in the future: #822.